sdks/python/apache_beam/ml/transforms/handlers.py

```diff
@@ -0,0 +1,436 @@
+      raw_data_feature_spec[key] = self._get_raw_data_feature_spec_per_column(
+          typ=value, col_name=key)
+    return raw_data_feature_spec
+
+  def convert_raw_data_feature_spec_to_dataset_metadata(
+      self, raw_data_feature_spec) -> dataset_metadata.DatasetMetadata:
+    raw_data_metadata = dataset_metadata.DatasetMetadata(
+        schema_utils.schema_from_feature_spec(raw_data_feature_spec))
+    return raw_data_metadata
+
+  def _get_raw_data_feature_spec_per_column(
+      self, typ: type, col_name: str) -> tf.io.VarLenFeature:
+    """
+    Return a FeatureSpec object to be used with
+    tft_beam.AnalyzeAndTransformDataset
+    Args:
+      typ: A type of the column.
+      col_name: A name of the column.
+    Returns:
+      A FeatureSpec object.
+    """
+    # lets conver the builtin types to typing types for consistency.
+    typ = native_type_compatibility.convert_builtin_to_typing(typ)
+    primitive_containers_type = (
+        list,
+        collections.abc.Sequence,
+    )
+    is_primitive_container = (
+        typing.get_origin(typ) in primitive_containers_type)
+
+    if is_primitive_container:
+      dtype = typing.get_args(typ)[0]  # type: ignore[attr-defined]
+      if len(typing.get_args(typ)) > 1 or typing.get_origin(dtype) == Union:  # type: ignore[attr-defined]
+        raise RuntimeError(
+            f"Union type is not supported for column: {col_name}. "
+            f"Please pass a PCollection with valid schema for column "
+            f"{col_name} by passing a single type "
+            "in container. For example, List[int].")
+    elif issubclass(typ, np.generic) or typ in _default_type_to_tensor_type_map:
+      dtype = typ
+    else:
+      raise TypeError(
+          f"Unable to identify type: {typ} specified on column: {col_name}. "
+          f"Please provide a valid type from the following: "
+          f"{_default_type_to_tensor_type_map.keys()}")
+    return tf.io.VarLenFeature(_default_type_to_tensor_type_map[dtype])
+
+  def get_raw_data_metadata(
+      self, input_types: Dict[str, type]) -> dataset_metadata.DatasetMetadata:
+    raw_data_feature_spec = self.get_raw_data_feature_spec(input_types)
+    return self.convert_raw_data_feature_spec_to_dataset_metadata(
+        raw_data_feature_spec)
+
+  def write_transform_artifacts(self, transform_fn, location):
+    """
+    Write transform artifacts to the given location.
+    Args:
+      transform_fn: A transform_fn object.
+      location: A location to write the artifacts.
+    Returns:
+      A PCollection of WriteTransformFn writing a TF transform graph.
+    """
+    return (
+        transform_fn
+        | 'Write Transform Artifacts' >>
+        transform_fn_io.WriteTransformFn(location))
+
+  def _fail_on_non_default_windowing(self, pcoll: beam.PCollection):
+    if not pcoll.windowing.is_default():
+      raise RuntimeError(
+          "TFTProcessHandler only supports GlobalWindows when producing "
```

Comment:
Nit: errors shouldn't explicitly reference `TFTProcessHandler` anymore, instead we should provide the transform name where we ran into this (or just the list of transform names)

Reply:
I changed this from TFTProcessHandler to MLTransform.